src/sparseml/optim/helpers.py

```diff
@@ -65,25 +64,25 @@ def load_recipe_yaml_str(
         stub to a SparseZoo model whose recipe will be downloaded and loaded.
         SparseZoo stubs should be preceded by 'zoo:', and can contain an optional
         '?recipe_type=<type>' parameter or include a `/<type>` subpath. Can also
-        be a SparseZoo Recipe object. i.e. '/path/to/local/recipe.yaml',
+        be a SparseZoo File object. i.e. '/path/to/local/recipe.md',
         'zoo:model/stub/path', 'zoo:model/stub/path?recipe_type=transfer_learn',
         'zoo:model/stub/path/transfer_learn'. Additionally, a raw
          yaml str is also supported in place of a file path.
     :param variable_overrides: dict of variable values to replace
         in the loaded yaml string. Default is None
     :return: the recipe YAML configuration loaded as a string
     """
-    if isinstance(file_path, Recipe):
+    if isinstance(file_path, File):
         # download and unwrap Recipe object
-        file_path = file_path.downloaded_path()
+        file_path = file_path.path
 
     if not isinstance(file_path, str):
         raise ValueError(f"file_path must be a str, given {type(file_path)}")
 
     if file_path.startswith("zoo:"):
         # download from zoo stub
-        recipe = Zoo.download_recipe_from_stub(file_path)
-        file_path = recipe.downloaded_path()
+        model = Model(file_path)
+        file_path = model.recipes.default.path
```

Comment:
does default take the query params into account?

Reply:
hell yeah.

Reply:
if no query params -> default is going to point to "original.md"
if recipe-specific query params -> default is going to point to the recipe specified in query params.

Reply:
this did not seem to be working + was missing backwards compatibility - added https://github.com/neuralmagic/sparsezoo/pull/189 to fix